File: StoneBC/AppConfig.swift
//
//  AppConfig.swift
//  StoneBC
//
//  Config-driven settings for open-source reuse by other bike co-ops
//

import Foundation

struct AppConfig: Codable {
    let coalitionName: String
    let shortName: String
    let tagline: String
    let websiteURL: String
    let email: String
    let phone: String?
    let instagramHandle: String?
    let location: LocationInfo?
    let colors: BrandColors
    let features: FeatureFlags
    let dataURLs: DataURLs?

    struct LocationInfo: Codable {
        let name: String
        let address: String
        let city: String
        let state: String
        let zip: String
    }

    struct BrandColors: Codable {
        let brandBlue: String
        let brandGreen: String
        let brandAmber: String
    }

    struct FeatureFlags: Codable {
        let enableMarketplace: Bool
        let enableCommunityFeed: Bool
        let enableRoutes: Bool
        let enableEvents: Bool
        let enableGallery: Bool
        let enableRadio: Bool
    }

    struct DataURLs: Codable {
        let wordpressBase: String?
        let bikes: String?
        let events: String?
        let posts: String?
    }

    struct APIKeys: Codable {
        let trailforksAppId: String?
        let trailforksAppSecret: String?
        let stravaClientId: String?
        let stravaClientSecret: String?
    }

    let apiKeys: APIKeys?

    static func load() -> AppConfig {
        guard let url = Bundle.main.url(forResource: "config", withExtension: "json"),
              let data = try? Data(contentsOf: url),
              let config = try? JSONDecoder().decode(AppConfig.self, from: data) else {
            return .default
        }
        return config
    }

    static let `default` = AppConfig(
        coalitionName: "Stone Bicycle Coalition",
        shortName: "SBC",
        tagline: "Building Community Through Cycling",
        websiteURL: "https://stonebicyclecoalition.com",
        email: "info@stonebicyclecoalition.com",
        phone: nil,
        instagramHandle: "stone_bicycle_coalition",
        location: LocationInfo(
            name: "Minneluzahan Senior Center",
            address: "315 N 4th St",
            city: "Rapid City",
            state: "SD",
            zip: "57701"
        ),
        colors: BrandColors(
            brandBlue: "#2563eb",
            brandGreen: "#059669",
            brandAmber: "#f59e0b"
        ),
        features: FeatureFlags(
            enableMarketplace: true,
            enableCommunityFeed: true,
            enableRoutes: true,
            enableEvents: true,
            enableGallery: true,
            enableRadio: true
        ),
        dataURLs: nil,
        apiKeys: nil
    )
}

File: android/app/src/main/kotlin/com/traddiff/stonebc/data/models/AppConfig.kt
package com.traddiff.stonebc.data.models

import kotlinx.serialization.Serializable

@Serializable
data class AppConfig(
    val coalitionName: String,
    val shortName: String,
    val tagline: String,
    val websiteURL: String,
    val email: String,
    val phone: String? = null,
    val instagramHandle: String? = null,
    val location: Location? = null,
    val colors: Colors = Colors(),
    val features: Features = Features(),
    val dataURLs: DataURLs = DataURLs(),
    val apiKeys: ApiKeys? = null
) {
    @Serializable
    data class Location(
        val name: String,
        val address: String,
        val city: String,
        val state: String,
        val zip: String
    )

    @Serializable
    data class Colors(
        val brandBlue: String = "#2563eb",
        val brandGreen: String = "#059669",
        val brandAmber: String = "#f59e0b"
    )

    @Serializable
    data class Features(
        val enableMarketplace: Boolean = true,
        val enableCommunityFeed: Boolean = true,
        val enableRoutes: Boolean = true,
        val enableEvents: Boolean = true,
        val enableGallery: Boolean = true,
        val enableRadio: Boolean = false,
        val enableWeather: Boolean = true
    )

    @Serializable
    data class DataURLs(
        val wordpressBase: String = "",
        val bikes: String = "",
        val events: String = "",
        val posts: String = ""
    )

    @Serializable
    data class ApiKeys(
        val trailforksAppId: String? = null,
        val trailforksAppSecret: String? = null,
        val stravaClientId: String? = null,
        val stravaClientSecret: String? = null,
        val openWeatherApiKey: String? = null
    )

    companion object {
        val Default = AppConfig(
            coalitionName = "Stone Bicycle Coalition",
            shortName = "SBC",
            tagline = "Building Community Through Cycling",
            websiteURL = "https://stonebicyclecoalition.com",
            email = "info@stonebicyclecoalition.com"
        )
    }
}
